src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectGrantController.java

```diff
@@ -65,167 +66,300 @@
+            @ApiParam(value = "The name of the bucket where the catalog objects are stored.", required = true) @PathVariable String bucketName,
+            @ApiParam(value = "The name of the object in the bucket.", required = true) @PathVariable String catalogObjectName)
             throws NotAuthenticatedException, AccessDeniedException {
-
         if (sessionIdRequired) {
+            AuthenticatedUser user = restApiAccessService.getUserFromSessionId(sessionId);
             // Check session validation
-            if (!restApiAccessService.isUserSessionActive(sessionId, currentUser)) {
+            if (!restApiAccessService.isUserSessionActive(sessionId, user.getName())) {
                 throw new AccessDeniedException("Session id is not active. Please login.");
             }
+            if (restApiAccessService.isAPublicBucket(bucketName)) {
+                throw new DataIntegrityViolationException("Bucket: " + bucketName +
+                                                          " is public. You can not assign a grant to it");
+            }
+            // Check Grants
+            if (!restApiAccessService.isBucketAccessibleByUser(sessionIdRequired, sessionId, bucketName)) {
+                if (!catalogObjectGrantService.checkInCatalogObjectGrantsIfTheUserOrUserGroupHasAdminRightsOverTheCatalogObject(user,
+                                                                                                                                bucketName,
+                                                                                                                                catalogObjectName)) {
+                    throw new CatalogObjectGrantAccessException(bucketName, catalogObjectName);
+                }
+            }
         }
-
-        return catalogObjectGrantService.getAllAssignedCatalogObjectGrantsForTheCurrentUserAndHisGroup(currentUser,
-                                                                                                       userGroup,
-                                                                                                       catalogObjectName,
-                                                                                                       bucketName);
-
+        return catalogObjectGrantService.getAllCreatedCatalogObjectGrantsForThisBucket(bucketName, catalogObjectName);
     }
 
     @SuppressWarnings("DefaultAnnotationParam")
-    @ApiOperation(value = "Get all created grants by the current user on all bucket's object")
+    @ApiOperation(value = "Delete all object grants inside a bucket")
     @ApiResponses(value = { @ApiResponse(code = 401, message = "User not authenticated"),
                             @ApiResponse(code = 403, message = "Permission denied"), })
-    @RequestMapping(method = GET)
+    @RequestMapping(value = REQUEST_API_QUERY, method = DELETE)
```

Comment:
I don't understand the request mapping here

Reply:
if it's for an object, then please change the method name and ApiOperation name to
deleteAllCatalogObjectGrants
Delete all grant associated with a catalog object

Reply:
Yes it is associated with an object. It deletes all grants associated to it (Like you've mentioned)